runner_service/utils.py

```diff
@@ -23,6 +23,10 @@
 class RunnerServiceError(Exception):
     pass
 
+def create_directory(dir_path):
+    """ Create directory if it doesn't exist """
+    if not os.path.exists(dir_path):
+        os.makedirs(dir_path)
```

Comment:
please try makedirs and handle exception instead, avoiding race conditions on directory creation